presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketHandle.java

```diff
@@ -62,11 +63,25 @@ public int getReadBucketCount()
 
     public HiveBucketProperty toTableBucketProperty()
     {
+        if (isVirtuallyBucketed(columns)) {
+            return null;
+        }
+
         return new HiveBucketProperty(
                 columns.stream()
                         .map(HiveColumnHandle::getName)
                         .collect(toList()),
                 tableBucketCount,
                 ImmutableList.of());
     }
+
+    public static HiveBucketHandle createVirtualBucketHandle(int virtualBucketCount)
+    {
+        return new HiveBucketHandle(ImmutableList.of(pathColumnHandle()), virtualBucketCount, virtualBucketCount);
+    }
+
+    public static boolean isVirtuallyBucketed(List<HiveColumnHandle> bucketColumns)
```

Comment:
Can this be a method for `BucketSplitInfo` ?

Reply:
If we move this method to `BucketSplitInfo` then we should make `HiveBucketHandle` only represent physical bucket and keep the virtual bucketing information in elsewhere. I am fine with either way.